app/scripts/router.js

```diff
@@ -244,9 +247,18 @@ function (
       // rendered for any reason, including if the view was
       // automatically redirected.
       var self = this;
-
       viewToShow.logScreen();
-      return viewToShow.render()
+      // perform any actions needed by the broker before the view has rendered
+      return self.broker.beforeRender()
```

Comment:
This seems strange to do here, on every screen render. Can this be done in app-start on initialization, or as part of the broker's fetch/initialize function?